docs/source/design/ui_design.rst

```diff
@@ -0,0 +1,182 @@
+######################
+User Interface Design
+######################
+
+This document outlines the design principles and guidelines for the user
+interface (UI) of NWChemEx. Our main goal is to provide a user-friendly and
+intuitive interface that enables users to carry out a variety of quantum
+chemistry calculations seamlessly. However, please note that a graphical user
+interface (GUI) is not part of the current design.
+
+The UI should enable users to conduct commonly used quantum chemistry
+calculations such as single-point energies and geometry optimizations in as few
+steps as possible. It should also be flexible enough to allow for more complex
+workflows using the helper/driver functions and PluginPlay modules. Users can
+execute these calculations either via Python scripts or interactively using
+Jupyter notebooks. For those who desire a fine-grained control of their
+workflow, they can utilize the Python and/or C++ API.
+
+Common Quantum Chemistry Calculations
+=======================================
+
+1. **Single Point Calculations**
+   
+   In these calculations the geometry of the molecule (arrangement of atomic
+   nuclei) is fixed. The user can be interested in the energy, wavefunction,
+   gradients (first order partial derivative of the energy with respect to nuclear
```

Comment:
I haven't heard much referring to gradients as single-point calculations. Not saying it's wrong, but it feels a little odd. Maybe add an additional list item for derivatives and mention both gradients and Hessians there.

Reply:
Good point and I agree this can cause confusion although it is not wrong but an unusual definition. I'll update the list.

Reply:
@jwaldrop107 could you check the updated [one](https://github.com/NWChemEx-Project/.github/blob/ui_design/docs/source/design/ui_design.rst#common-quantum-chemistry-calculations).